pkg/kubelet/kubelet_getters.go

```diff
@@ -235,6 +237,15 @@ func (kl *Kubelet) GetNode() (*v1.Node, error) {
 	if kl.kubeClient == nil {
 		return kl.initialNode(context.TODO())
 	}
+	// if we have a valid kube client, we wait for initial lister to sync
+	if !kl.nodeHasSynced() {
+		err := wait.PollImmediate(time.Second, maxWaitForAPIServerSync, func() (bool, error) {
+			return kl.nodeHasSynced(), nil
+		})
+		if err != nil {
+			return nil, fmt.Errorf("nodes have not yet been read at least once, cannot construct node object")
+		}
+	}
```

Comment:
this poll is problematic.
GetNode is called in a loop, which means that the poll is present on each call.

the sequence looks like:
- GetNode()
  - Poll for nodeHasSynced()
  - Poll ....
- GetNode()
  - Poll for nodeHasSynced()
  - Poll ....

if there is a valid client, arguably the informer sync check should be outside of this loop.

Reply:
Wouldn't it be enough to check `HasSynced` once before creating the Kubelet object?

Reply:
it would only sync after the api server is up.

Reply:
if the API server is down, there is nothing to sync. Am I missing something?

Reply:
BTW we are discussion how to improve this in:
https://github.com/kubernetes/kubernetes/pull/99336

if the kubelet is managing the first server instance in the cluster as a static pod and if the HasSynced check is before the kubelet object creation this means for such a kubelet instance the single HasSynced check will always be false.

maybe that's what we want, given this is the first node in the cluster and there is no need to sync it.

for subsequent kubelets the check should pass.